storage/sftp/storage_sftp.go

```diff
@@ -0,0 +1,161 @@
+/*
+ * knoxite
+ *     Copyright (c) 2019, Fabian Siegel <fabians1999@gmail.com>
+ *
+ *   For license see LICENSE
+ */
+
+package sftp
+
+import (
+	"errors"
+	"io/ioutil"
+	"net"
+	"net/url"
+	"os"
+	"os/user"
+	"path/filepath"
+
+	"github.com/pkg/sftp"
+	"golang.org/x/crypto/ssh"
+	agent "golang.org/x/crypto/ssh/agent"
+	kh "golang.org/x/crypto/ssh/knownhosts"
+
+	knoxite "github.com/knoxite/knoxite/lib"
+)
+
+type StorageSFTP struct {
+	url   url.URL
+	ssh   *ssh.Client
+	sftp  *sftp.Client
+	login bool
+	knoxite.StorageFilesystem
+}
+
+var (
+	ErrInvalidAuthentication = errors.New("Wrong Username or Password")
```

Comment:
This seems to be unused by now. Generally speaking, we should try to avoid declaring error types in a storage backend, and instead use errors defined by `knoxite`, as it can handle them appropriately.